pyscf/mcpdft/test/test_mgga.py

```diff
@@ -0,0 +1,208 @@
+        dr[i] = (step #* numpy.sin((i+1)*step)
+                * ln2*(1+x)**.6 *(-.6/(1+x)*np.log((1-x)/2)+1/(1-x)))
+    return r[::-1], dr[::-1]
+
+my_grids = {'atom_grid': (99,590),
+        'radi_method': om_treutler_ahlrichs,
+        'prune': False,
+        'radii_adjust': None}
+
+def get_lih (r, stateaverage=False, functional='tM06L', basis='sto3g'):
+    mol = gto.M (atom='Li 0 0 0\nH {} 0 0'.format (r), basis=basis,
+                 output='/dev/null', verbose=0)
+    mf = scf.RHF (mol).run ()
+    if functional == 'tM06L0':
+        tM06L0 = 't' + mcpdft.hyb('M06L',0.25, hyb_type='average')
+        mc = mcpdft.CASSCF(mf, tM06L0, 5, 2, grids_attr=my_grids)
+    else:
+        mc = mcpdft.CASSCF(mf, functional, 5, 2, grids_attr=my_grids)
+
+    if stateaverage:
+        mc = mc.state_average_([0.5, 0.5])
+
+    mc.fix_spin_(ss=0)
+    mc = mc.run()
+    return mc
+
+def get_water_triplet(functional='tM06L', basis='6-31G'):
+    mol = gto.M(atom='''
+ O     0.    0.000    0.1174
+ H     0.    0.757   -0.4696
+ H     0.   -0.757   -0.4696
+    ''',basis=basis, spin=2,output='/dev/null', verbose=0)
+
+    mf = scf.RHF(mol).run()
+
+    mc = mcpdft.CASSCF(mf, functional, 2, 2, grids_attr=my_grids)
+    solver1 = fci.direct_spin1.FCI(mol)
+    solver1 = fci.addons.fix_spin(solver1, ss=2)
+    mc.fcisolver = solver1
+    mc = mc.run()
+    return mc
+
+def setUpModule():
+    global get_lih, lih_tm06l, lih_tmc23, lih_tm06l_sa2, lih_tmc23_sa2,lih_tm06l0
+    global get_water_triplet, water_tm06l, water_tmc23
+    lih_tm06l = get_lih(1.5, functional='tM06L')
+    lih_tmc23 = get_lih(1.5, functional='tMC23')
+    lih_tm06l_sa2 = get_lih(1.5, stateaverage=True, functional='tM06L')
+    lih_tmc23_sa2 = get_lih(1.5, stateaverage=True, functional='tMC23')
+    lih_tm06l0 = get_lih(1.5, functional='tM06L0')
+    water_tm06l = get_water_triplet()
+    water_tmc23 = get_water_triplet(functional='tMC23')
+
+def tearDownModule():
+    global lih_tm06l, lih_tmc23, lih_tm06l_sa2, lih_tmc23_sa2
+    global lih_tm06l0, water_tm06l, water_tmc23
+    lih_tm06l.mol.stdout.close()
+    lih_tmc23.mol.stdout.close()
+    lih_tm06l_sa2.mol.stdout.close()
+    lih_tmc23_sa2.mol.stdout.close()
+    lih_tm06l0.mol.stdout.close()
+    water_tm06l.mol.stdout.close()
+    water_tmc23.mol.stdout.close()
+    del lih_tm06l, lih_tmc23, lih_tm06l_sa2, lih_tmc23_sa2
+    del lih_tm06l0, water_tm06l, water_tmc23
+
+class KnownValues(unittest.TestCase):
+
+    def assertListAlmostEqual(self, first_list, second_list, expected):
+        self.assertTrue(len(first_list) == len(second_list))
+        for first, second in zip(first_list, second_list):
+            self.assertAlmostEqual(first, second, expected)
+
+    def test_tmgga(self):
+        e_mcscf = lih_tm06l.e_mcscf
+        epdft = lih_tm06l.e_tot
+
+        sa_e_mcscf = lih_tm06l_sa2.e_mcscf
+        sa_epdft = lih_tm06l_sa2.e_states
+
+        E_CASSCF_EXPECTED = -7.88214917
```

Comment:
Generally I am in favor of including the git commit from which these expected values as a comment. Also it is good to include a comment for if a number came from OpenMolcas or PySCF.

Reply:
I have included at the top of file that these reference values are generated from OpenMolcas, and included the version and other relevant information.

Should I include that information in-front of each reference value?

Reply:
I would. The reference values at the top will be forgotten about if others add new tests later in the file which might not come from OpenMolcas anymore. Or if the values start to change over time (for whatever reason). Best to keep the comments as close to the reference values as possible.

Reply:
Done.